common/main.c

```diff
@@ -5,19 +5,36 @@
 
 #include <compiler.h>
 #include <console.h>
+#include <debug.h>
 #include <dm.h>
 #include <stdbool.h>
 #include <work.h>
+#include <drivers/watchdog.h>
 #include <platform/devices.h>
 
+#define WDOG_TIMEOUT (5 * 1000 * 1000) /* 5 seconds */
+
 noreturn void main(void);
 
+void main(void);
+
 noreturn void
 main(void)
 {
+	struct device *twd_dev;
+
 	console_init(DEV_UART0);
 	dm_init();
 
+	/* Enable trusted watchdog. */
+	if ((twd_dev = dm_get_by_class(DM_CLASS_WATCHDOG))) {
+		watchdog_enable(twd_dev, WDOG_TIMEOUT);
+		info("Trusted watchdog enabled.");
+	} else {
+		warn("Trusted watchdog did not start.");
```

Comment:
This message is misleading. It's in the case no watchdog was found (which is programmer error, not user error, and we really don't have space for extra strings), not when one was found and it failed.

Reply:
I disagree that the message is _misleading_; this message is certainly _accurate_ (the watchdog will not start if the device cannot be found). Although this may be helpful to know, I'll remove the message to save space.

Reply:
Well, yes, it's *accurate*, but from my point of view it implies that you attempted to start it, and that somehow failed, while in reality no attempt was made. (Not that it matters, but that's why it was misleading to me.)

Reply:
Thanks for the explanation; I see where you're coming from. If we decide to add this again further down the line, a more representational message should be used.